acme/acme.go

```diff
@@ -331,6 +334,14 @@ func (a *ACME) CreateLocalConfig(tlsConfig *tls.Config, checkOnDemandDomain func
 func (a *ACME) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
 	domain := types.CanonicalDomain(clientHello.ServerName)
 	account := a.store.Get().(*Account)
+	//use regex to test for wildcard certs that might have been added into TLSConfig
+	for k := range a.TLSConfig.NameToCertificate {
+		selector := "^" + strings.Replace(k, "*.", ".*\\.?", -1) + "$"
```

Comment:
My concern is for the possibility of not having SANs. That is, I have a wildcard for *.mydomain.com. It will not cover *.test.mydomain.com, and I don't want to add SANs for that. Instead I want to use the wildcard for *.mydomain.com, but have ACME generate certificates for say bar.foo.mydomain.com.
Will the code as-is allow this, or will it attempt to use the wildcard even though it doesn't apply due to the lack of a SAN?